### PR TITLE
Seperate `filter` and `description` arguments in `EditorFileDialog/FileDialog.add_filter()`

### PR DESCRIPTION
**Pull request by fire-forge**
FileDialog/EditorFileDialog filters have both a file name filter and a description. Previously `EditorFileDialog/FileDialog.add_filter()` required these to be contained in the same string using `;` as a separator, but this PR changes it to take two arguments, the first being the filter and the second being an optional description.

This does not break compatibility because the old syntax still works. Behind the scenes `add_filter()` just combines the arguments into the `"filter ; description"` format if a description is given. If only a filter is given and no description, then it will add the filter as-is to the filters array, which is identical to the old behavior.

#### Example:
```gdscript
# These lines are equivalent:
add_filter("*.png ; PNG") # Old syntax
add_filter("*.png", "PNG") # New syntax

```

While the old format does still work, I changed all usages of `add_filter()` in the codebase to use the new syntax for consistency.

### CODE REVIEW

File: editor/editor_file_dialog.cpp [hunk 1]
```diff
@@ -1218,6 +1222,7 @@ void EditorFileDialog::_update_drives(bool p_select) {
 
 void EditorFileDialog::_update_icons() {
 	// Update icons.
+	file->set_right_icon(item_list->get_theme_icon(SNAME("Search"), SNAME("EditorIcons")));
```
Comment (KoBeWi) [inline editor/editor_file_dialog.cpp:1225]: Why this change?
This is not even search bar.

File: editor/editor_audio_buses.cpp [hunk 2]
```diff
@@ -1332,7 +1332,7 @@ EditorAudioBuses::EditorAudioBuses() {
 	List<String> ext;
 	ResourceLoader::get_recognized_extensions_for_type("AudioBusLayout", &ext);
 	for (const String &E : ext) {
-		file_dialog->add_filter(vformat("*.%s; %s", E, TTR("Audio Bus Layout")));
+		file_dialog->add_filter(vformat("*.%s", E), TTR("Audio Bus Layout"));
```
Comment (KoBeWi) [inline editor/editor_audio_buses.cpp:1335]:
```suggestion
		file_dialog->add_filter("*." + E, TTR("Audio Bus Layout"));
```
Could be like this probably.

File: editor/plugins/theme_editor_plugin.cpp [hunk 3]
```diff
@@ -2080,7 +2080,7 @@ ThemeItemEditorDialog::ThemeItemEditorDialog(ThemeTypeEditor *p_theme_type_edito
 	List<String> ext;
 	ResourceLoader::get_recognized_extensions_for_type("Theme", &ext);
 	for (const String &E : ext) {
-		import_another_theme_dialog->add_filter(vformat("*.%s; %s", E, TTR("Theme Resource")));
+		import_another_theme_dialog->add_filter(vformat("*.%s", E), TTR("Theme Resource"));
```
Comment (KoBeWi) [inline editor/plugins/theme_editor_plugin.cpp:2083]: Same as with audio bus layout. This can be changed to match the style of other filter definitions.

File: editor/plugins/theme_editor_plugin.cpp [hunk 4]
```diff
@@ -3667,7 +3667,7 @@ ThemeEditor::ThemeEditor() {
 	List<String> ext;
 	ResourceLoader::get_recognized_extensions_for_type("PackedScene", &ext);
 	for (const String &E : ext) {
-		preview_scene_dialog->add_filter(vformat("*.%s; %s", E, TTR("Scene")));
+		preview_scene_dialog->add_filter(vformat("*.%s", E), TTR("Scene"));
```
Comment (KoBeWi) [inline editor/plugins/theme_editor_plugin.cpp:3670]: Same.

File: editor/project_export.cpp [hunk 5]
```diff
@@ -891,7 +891,7 @@ void ProjectExportDialog::_export_project() {
 	List<String> extension_list = platform->get_binary_extensions(current);
 	for (int i = 0; i < extension_list.size(); i++) {
 		// TRANSLATORS: This is the name of a project export file format. %s will be replaced by the platform name.
-		export_project->add_filter(vformat("*.%s; %s", extension_list[i], vformat(TTR("%s Export"), platform->get_name())));
+		export_project->add_filter(vformat("*.%s", extension_list[i]), vformat(TTR("%s Export"), platform->get_name()));
```
Comment (KoBeWi) [inline editor/project_export.cpp:894]: Same.

File: editor/editor_file_dialog.cpp (hunk 1 above)
Comment (fire-forge) [reply editor/editor_file_dialog.cpp:1225]: Honestly I'm not sure why I did that. I remember thinking that it was a search bar, but clearly it isn't. I must have been tired or something lol.

**Review by KoBeWi — APPROVED** [review]
Looks good.

It could be interesting to expose the filters in the inspector in this way, i.e. in filter/description pairs.
